GiniVision/Classes/GiniPDFDocument.swift

```diff
@@ -29,10 +29,11 @@ final public class GiniPDFDocument: NSObject, GiniVisionDocument {
     
     public init(data:Data) {
         self.data = data
-        self.previewImage = UIImageNamedPreferred(named: "cameraDefaultDocumentImage") // Here should be the first rendered page
-        if let dataProvider = CGDataProvider(data: data as CFData) {
-            let pdfDocument = CGPDFDocument(dataProvider)
-            self.numberPages = pdfDocument?.numberOfPages ?? 0
+        super.init()
+        
+        if let dataProvider = CGDataProvider(data: data as CFData), let pdfDocument = CGPDFDocument(dataProvider) {
+            self.numberPages = pdfDocument.numberOfPages
+            self.previewImage = renderFirstPage(fromPdf: pdfDocument)
```

Comment:
Can this be "lazy"? You don't need the preview right from the initialise.

Reply:
I think it depends, because for instance when you use "Open with..." method you need the preview image asap. So I would keep this here (also it is useful since CGPDFDocument is used for the number of pages and it can be used to render the first image)